File: api/proxy_wol.py
# proxy_wol.py

import requests
from flask import Flask, jsonify, request

app = Flask(__name__)

@app.route('/api/proxy-wol')
def proxy_wol():
    year = request.args.get('year')
    week = request.args.get('week')
    url = f'https://wol.jw.org/es/wol/meetings/r4/lp-s/{year}/{week}'

    try:
        response = requests.get(url)
        response.raise_for_status()
        return response.content, response.status_code
    except requests.exceptions.RequestException as e:
        return jsonify({'error': 'Failed to fetch data from WOL'}), 500

if __name__ == '__main__':
    app.run(debug=True)

File: api/proxy-wol.js
const axios = require('axios');

module.exports = async (req, res) => {
  const { year, week } = req.query;
  const url = `https://wol.jw.org/es/wol/meetings/r4/lp-s/${year}/${week}`;

  try {
    const response = await axios.get(url);
    res.status(200).send(response.data);
  } catch (error) {
    res.status(500).json({ error: 'Failed to fetch data' });
  }
};
